app/Models/BeatmapSet.php

```diff
@@ -471,6 +473,177 @@ public function replies($time = null)
         return $new;
     }
 
+    public static function imageSizes()
+    {
+        $shapes = ['cover', 'card', 'list'];
+        $scales = ['', '@2x'];
+
+        $sizes = [];
+        foreach ($shapes as $shape) {
+            foreach ($scales as $scale) {
+                $sizes[] = "$shape$scale";
+            }
+        }
+
+        return $sizes;
+    }
+
+    public function allImageURLs()
+    {
+        $urls = [];
+        foreach (self::imageSizes() as $size) {
+            $urls[$size] = $this->coverImageURL($size);
+        }
+
+        return $urls;
+    }
+
+    public static function isValidCoverSize($coverSize)
+    {
+        $validSizes = array_merge(['raw', 'fullsize'], self::imageSizes());
+
+        return in_array($coverSize, $validSizes, true);
+    }
+
+    public function coverImageURL($coverSize = 'cover')
+    {
+        if (!self::isValidCoverSize($coverSize)) {
+            return false;
+        }
+
+        $timestamp = 0;
+        if ($this->cover_updated_at) {
+            $timestamp = $this->cover_updated_at->format('U');
+        }
+
+        return $this->storage()->url("/beatmaps/{$this->beatmapset_id}/covers/{$coverSize}.jpg?{$timestamp}");
```

Comment:
maybe the `"/beatmaps/{$this->beatmapset_id}/covers/` part worth making a function as well? (`coverDir()` or something like that).